extensions/producthunt/src/util/textUtils.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * Utility functions for text processing and sanitization
+ */
+
+/**
+ * Cleans text by replacing common Unicode escapes and HTML entities with their actual characters
+ * @param text The text to clean
+ * @returns The cleaned text
+ */
+export function cleanText(text: string | undefined | null): string {
+  if (!text) return "";
+
+  // Define replacement maps for more efficient processing
+  const commonUnicodeEscapes: Record<string, string> = {
+    "\\u0026": "&",
+    "\\u003c": "<",
+    "\\u003e": ">",
+    "\\u0022": '"',
+    "\\u0027": "'",
+  };
+
+  const htmlEntities: Record<string, string> = {
+    "&amp;": "&",
+    "&lt;": "<",
+    "&gt;": ">",
+    "&quot;": '"',
+    "&#39;": "'",
+  };
+
+  // Replace unicode escapes
+  let cleanedText = text;
+  Object.entries(commonUnicodeEscapes).forEach(([escape, char]) => {
+    cleanedText = cleanedText.replace(new RegExp(escape, "g"), char);
+  });
```

Comment:
As I mentioned in my last comment, I still have some \u0026 displayed. A way to fix it is by adding the code after the first unicode escapes.

```suggestion
  // Define replacement maps for more efficient processing
  const htmlEntities: Record<string, string> = {
    "&amp;": "&",
    "&lt;": "<",
    "&gt;": ">",
    "&quot;": '"',
    "&#39;": "'",
  };

  // Replace unicode escapes
  let cleanedText = text;

  cleanedText = cleanedText.replace(/\\u[0-9a-fA-F]{4}/g, (match) => {
    const hex = match.slice(2); // remove leading "\u"
    return String.fromCharCode(parseInt(hex, 16));
  });
```

Reply:
I was given an even better approach that:

- Eliminates the intermediate match variable
- Uses regex groups to directly access the hex value
- Is more compact while maintaining clarity

```
export function cleanText(text: string | undefined | null): string {
  if (!text) return "";

  return text
    // Convert all Unicode escapes (like \u0026)
    .replace(/\\u([0-9a-fA-F]{4})/g, (_, hex) => 
      String.fromCharCode(parseInt(hex, 16))
    )
    // Replace common HTML entities
    .replace(/&(amp|lt|gt|quot|#39);/g, (_, entity: 'amp'|'lt'|'gt'|'quot'|'#39') => 
      ({ amp: '&', lt: '<', gt: '>', quot: '"', '#39': "'" })[entity] || _
    );
}
```